R/class-workbook.R

```diff
@@ -4126,11 +4142,88 @@ wbWorkbook <- R6::R6Class(
       value
     },
 
+    validate_new_sheet = function(sheet) {
+      # returns nothing, throws error if there's a problem.
+
+      if (is.na(sheet)) {
+        stop("sheet cannot be NA")
+      }
+
+      if (is.numeric(sheet)) {
+        if (sheet %% 1 != 0) {
+          stop("If sheet is numeric it must be an integer")
+        }
+
+        if (sheet <= 0) {
+          stop("if sheet is an integer it must be a positive number")
+        }
+
+        if (sheet <= length(self$sheet_names)) {
+          stop("there is already a sheet at index ", sheet)
+        }
+
+        return()
+      }
+
+      sheet <- as.character(sheet)
+      if (any_illegal_chars(sheet)) {
+        stop("Illegal characters found in sheet. Please remove. See ?openxlsx::clean_worksheet_name")
+      }
+
+      if (nchar(sheet) > 31) {
+        stop("sheet names must be <= 32 chars")
+      }
+
+      if (tolower(sheet) %in% self$sheet_names) {
+        stop("a sheet with name '", sheet, '"already exists"')
+      }
+    },
+
+    get_sheet = function(sheet) {
```

Comment:
maybe `get_sheet_idx`?

Reply:
`get_sheet_index()`?  That would be more clear.

We could then have a `get_sheet_name()` that functions similarly (passed a number returns the name, passed the name and it checks that it's a valid one) -- always returns a character.

Reply:
Agreed, I tend to shorten such names, but that's just personal preference. Regarding names, I'm not really sure that we use sheet names ... yet(?)